screens/AppNavigator/screens/TransactionsScreen/screens/reducer.ts

```diff
@@ -0,0 +1,63 @@
+import { AddressActivity } from '@defichain/whale-api-client/dist/api/address'
+import { NavigationProp, ParamListBase } from '@react-navigation/native'
+import BigNumber from 'bignumber.js'
+
+// VM for ViewModel
+export interface VMTransaction {
+  id: string
+  desc: string // of each transaction type, eg: Sent, Add Liquidity
+  iconName: string
+  color: string
+  amount: string
+  block: number
+  token: string
+  txid: string
+}
+
+export function activitiesToViewModel (activities: AddressActivity[], navigation: NavigationProp<ParamListBase>): VMTransaction[] {
```

Comment:
Didn't actually done any automated test for FE code, I should start here?
(these functions actually only used by the transaction listing page, only the view model is shared)

@fuxingloh @thedoublejay